Helpers/PaymentValidation.php

```diff
@@ -393,4 +390,47 @@ public function cancelOrder(Phrase $internalError, bool $transitionOrder, Order
             $this->orderHelper->cancel($order->getId());
         }
     }
+
+    /**
+     * @param Order $order
+     *
+     * @return bool
+     */
+    protected function expireOrder(Order $order): bool
+    {
+        $order = $this->addCommentToOrder($order, __('Alma Payment is expired'));
+        $this->orderHelper->save($order);
+        $this->orderHelper->cancel($order->getId());
+        $this->logger->info('Payment is expired cancel Order :', [$order->getIncrementId()]);
+        return true;
+    }
+
+    /**
+     * @param Order $order
+     *
+     * @return OrderPaymentInterface
+     * @throws AlmaPaymentValidationException
+     */
+    public function getPayment(Order $order): OrderPaymentInterface
+    {
+        $payment = $order->getPayment();
+        if (!$payment) {
+            $internalError = __("Cannot get payment information from order %s", $order->getIncrementId());
+
+            $this->logger->error($internalError->render());
+            $this->addCommentToOrder($order, $internalError);
+            throw new AlmaPaymentValidationException($internalError->render());
+        }
+        return $payment;
+    }
+
+    /**
+     * @param AlmaPayment $almaPayment
+     *
+     * @return bool
+     */
+    private function paymentIsExpired(AlmaPayment $almaPayment): bool
+    {
+        return (bool)$almaPayment->expired_at;
```

Comment:
and what will happen if `expired_at` is after `now()` 😛 (this is not supposed to happen, but could we really be confident with what an API should return us ?)